### PR TITLE
move forumlist addon to core

### PR DESCRIPTION
**Pull request by rabuzarus**
Now the forumlist addon has been moved from addon to /include/forums.php

Some remarks:
* If the user is not connected with any forum the widget won't appear (I think this is nicer than to have a widget which says "No forum subscriptions")
* The value for listed forums is set to 10. If there are more than 10 forums, a "more" link appears to expand the list
* As I said in the support forum, I'm not really happy with the place for setting of forumlist_profile_advanced (maby we should think about it. suggestions are welcome. I would like to have it in edit advanced profile page but the settings should go to pconfig table -> I don't know how to solve this)

To-Do (would like to do this after this PR is accepted)
* delete forumlist addon from the Addon repository (Is there a way to disable the forumlist Addon in the database automatically on update?)
* find a way to have default values (widget_forumlist should be default on while forumlist_profile_advanced should be default off). In present state it should be OFF
* `SORT BY last-item` doesn't work
* integrate a badge which shows the unread posts of every forum

### CODE REVIEW

**Comment by annando** [comment]
Concerning your points:
* We already had had code to deactivate addons. Have a look at "statistics_json". This we had moved into the core as well. It deactivates itself.
* Concerning "last-item": I will make a change in the core for that. Could you file an issue?

File: include/identity.php [hunk 1]
```diff
@@ -603,6 +604,13 @@ function advanced_profile(&$a) {
 		if($txt = prepare_text($a->profile['work'])) $profile['work'] = array( t('Work/employment:'), $txt);
 
 		if($txt = prepare_text($a->profile['education'])) $profile['education'] = array( t('School/education:'), $txt );
+	
+		//show subcribed forum if it is enabled in the usersettings
+		if (feature_enabled($uid,'forumlist_profile')) {
+			require_once('include/forums.php');
```
Comment (annando) [inline include/identity.php]: Could you move the include to the header? I think this is cleaner.

Comment (rabuzarus) [inline include/identity.php]: > Could you move the include to the header? I think this is cleaner.

Should I move it to the top of the file or the top of function advanced_profile(&$a)?

Comment (fabrixxm) [inline include/identity.php]: I would say "top of the file"

**Comment by rabuzarus** [comment]
I have looked in the the nodeinfo.php there is  a function nodeinfo_plugin_enabled( ). This function checks if a plugin is enabled? And this function checks everytime if the statistics plugin is enabled.

But I think it would be cleaner to have something like a update routine. Something like the update which is done when the database sheme alters.

This routine would contain:
1.) compare the friendica versions
2.) run versions specific code if it would be existent
3.) In this case: delete the forumlist plugin form /addon and delete any old database entries of the old forumlist addon (and maybe set new one)

I think this routine could be also very useful in other cases.
@annando didn't you said that you are the backend guy :-D

**Comment by annando** [comment]
We have an update function. Just check the update.php.

**Comment by rabuzarus** [comment]
I have looked at it and I will try to use it. But (like always) I have some questions.

1.) I would like to to rename nodinfo_plugin_enabled() to plugin_enabled() and would like to move it to a more common place. Maybe plugins.php. Or would you suggest some other place?

2.) I have thought about a function to delete directories (for deleting addon/forumlist/) but it gives me stomach ache because of the security issues. Any suggestions how to solve this issue?

3.) dbstructure.php is only for altering the db scheme. Am I right? This would mean that I have to insert my code into update.php. Right?

**Comment by annando** [comment]
Concerning the first point: I would see it the same.

Second: Never ever delete the directory. All changes have to be done in the addons repository. But we should do it like the statistics addon. If we add some text in the header of the file then it will not be shown anymore.

Third: Yeah. You need to put it into the update.php. And you need to find the correct database structure number and you have to increase it. It's slightly complicated :)

**Comment by rabuzarus** [comment]
do I need to raise the db number in dbstructure.php to the one of boot.php even if I don't change something at the db scheme?

**Comment by annando** [comment]
Yeah. If you don't do it then it won't be executed. And AFAIK the number of the update function must be one lower than the new version number.

**Comment by rabuzarus** [comment]
@annando 
Can you please have a look at https://github.com/rabuzarus/friendica/tree/dbupdate_forumlist

I have tried to do a update function (have a look on update.php). Most of the code is borrowed from nodeinfo.php.

But it doesn't really work and I don't know why. I have done plenty of tests but I can't solve it. The update aborts. In most cases forumlist gets deleted from the addon table. But in it stays in the hooks table and in the config table (system, addons). 
log.out gives me 
`plugin.php:86:reload_plugins   Reloading plugin: forumlist`
or
`2015-11-12 02:35:56@ [NORMAL]:plugin.php:9:uninstall_plugin  Addons: uninstalling forumlist
2015-11-12 02:35:57@8ukvjh1t72citpd8lp8bnuv6qi9bvelb    [NORMAL]:plugin.php:27:install_plugin   Addons: installing forumlist`

**Comment by annando** [comment]
The code looks good. I really don't know why it is reinstalling the addon. There are different processes involved. All processes with text after the @ are called via web. So a background process is uninstalling it and a web process is re-enabling it.

Why? I don't know.

**Comment by rabuzarus** [comment]
Puhhh after long, long time the solution is is to delete the db entries manually. This seems to work. My tests were successful.

Are there any things left for accepting the PR?

**Comment by rabuzarus** [comment]
BTW: If an update fails an email should inform the admin, but it doesn't (https://github.com/friendica/friendica/blob/master/include/dbstructure.php#L8)

File: include/dbstructure.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 require_once("boot.php");
 require_once("include/text.php");
 
-define('NEW_UPDATE_ROUTINE_VERSION', 1170);
+define('NEW_UPDATE_ROUTINE_VERSION', 1191);
```
Comment (annando) [inline include/dbstructure.php]: This one must stay at "1170"

**Pull request merged**